data-gen-sdc.js

```diff
@@ -0,0 +1,165 @@
+  { id: '082', ticker: 'AMAT', company: 'Applied Materials'},
+  { id: '083', ticker: 'YETI', company: 'YETI'},
+  { id: '084', ticker: 'EA', company:   'EA'},
+  { id: '085', ticker: 'QCOM', company: 'Qualcomm'},
+  { id: '086', ticker: 'TGT', company:  'Target'},
+  { id: '087', ticker: 'TEVA', company: 'Teva Pharmaceutical'},
+  { id: '088', ticker: 'JNJ', company:  'Johnson & Johnson'},
+  { id: '089', ticker: 'IIPR', company: 'Innovative Industrial Properties'},
+  { id: '090', ticker: 'ACB', company:  'Aurora Cannabis'},
+  { id: '091', ticker: 'GE', company:   'GE' },
+  { id: '092', ticker: 'AAPL', company: 'Apple' },
+  { id: '093', ticker: 'F', company:    'Ford' },
+  { id: '094', ticker: 'CRON', company: 'Cronos Group'},
+  { id: '095', ticker: 'MSFT', company: 'Microsoft' },
+  { id: '096', ticker: 'GPRO', company: 'GoPro' },
+  { id: '097', ticker: 'FIT', company:  'Fitbit' },
+  { id: '098', ticker: 'AMD', company:  'AMD'},
+  { id: '099', ticker: 'FB', company:   'Facebook' },
+  { id: '100', ticker: 'CGC', company:  'Canopy Growth'}
+];
+
+const generateTickers = digits => {
+  const allTickers = companyData.map(val => val.ticker);
+  const validator = new Set(allTickers);
+  const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
+  (function loop(base, i){
+    for (let k = 0; k < possible.length; k++) {
+      if (i > 1) loop( base + possible[k], i-1);
+      else if (!validator.has(base + possible[k])) {
+        allTickers.push( base + possible[k]);
+      } 
+    }
+  })("", digits);
+  return allTickers;
+};
+
+const tickers = generateTickers(5);
+var j = 0;
+
+const generateSingleRecord = () => {
+  const descriptor = faker.company.catchPhraseDescriptor();
+  const descriptor2 = faker.lorem.sentence();
+  const material = faker.commerce.productMaterial();
+  const bs = faker.company.bs();
+  let data = `${tickers[j]}, ${faker.random.number(20)}, ${faker.random.number(20)}, ${faker.random.number(20)},
+  '${material} ${bs} ${tickers[j]} ${descriptor}. \n The ${bs} ${descriptor2} \n Overall, ${bs} ${tickers[j]} ${descriptor} ${descriptor2}',
+  '${material} ${bs} ${descriptor} ${tickers[j]}. \n For ${bs} ${descriptor2} \n Hence, ${bs} ${tickers[j]} ${descriptor} ${descriptor2}'
+  \n`;
+  j++;
+  return data;
+};
+
+const writeOneMillionTimes = (writer, data, encoding, callback) => {
+  let i = 1000000;
+  write();
+  function write() {
+    let ok = true;
+    do {
+      i--;
+      if (i === 0) {
+        // last item
+        data = generateSingleRecord();
+        writer.write(data, encoding, callback);
+      } else {
+        // See if we should continue, or wait.
+        data = generateSingleRecord();
+        ok = writer.write(data, encoding);
+      }
+    } while (i > 0 && ok);
+    if (i > 0) {
+      // had to stop early, bucket full, write some more once it drains
+      writer.once('drain', write);
+    }
+  };
+};
+
+let writer = fs.createWriteStream('./dataTest1000000.csv');
```

Comment:
If you plan to change the number of records that it is writing, I'd recommend tying line 140, 163, and any other references to the same variable to allow you to specify that number in one place and have it update the function and filename!